src/main/java/yahoo/YahooPortfolioData.java

```diff
@@ -103,13 +106,17 @@ public void addLot(int stockRow) {
      * Enters the transaction date for a stock. The transaction date is 
      * entered in the final transaction row of a stock.
      * @param stockRow int The Row number of the stock to enter the transaction value for
-     * @param date String The raw value of the date to be entered without seperators 
-     * e.g. 121020 is 12/10/20
+     * @param date LocalDate The LocalDate object 
      */
-    private void enterDate(int stockRow, String date) {
+    private void enterDate(int stockRow, LocalDate date) {
         By dateInputPath = By.xpath("//table/tbody[" + stockRow + "]/tr[3]/td/table/tbody/tr[last()-1]/td[1]/input[@type='date']");
         WebElement dateInput = driver.findElement(dateInputPath);
-        dateInput.sendKeys(date);
+
+        String value = date.format(DateTimeFormatter.ofPattern("ddMMyyyy"));
+        dateInput.sendKeys(value + Keys.TAB);
+
+        // The date value is saved in a different format
+        this.checkEntry(dateInput, date.format(DateTimeFormatter.ofPattern("yyyy-MM-dd")), stockRow);
```

Comment:
Implement the Check entry functionality within this method, as entering a date behaves differently to a text. The value which is entered and the value being saved have different formats and the current check entry method isn't capable of doing that.